packages/collector/test/tracing/cloud/aws/s3/test.js

```diff
@@ -0,0 +1,200 @@
+
+              const response = await appControls.sendRequest({
+                method: 'GET',
+                path: `${apiPath}${withErrorOption}`,
+                simple: withError === false
+              });
+
+              return verify(appControls, response, apiPath, withError);
+            });
+          });
+        });
+      });
+    });
+
+    function verify(controls, response, apiPath, withError) {
+      return retry(() => {
+        return agentControls.getSpans().then(spans => verifySpans(controls, spans, apiPath, withError));
+      }, retryTime);
+    }
+
+    function verifySpans(controls, spans, apiPath, withError) {
+      const httpEntry = verifyHttpEntry(spans, apiPath, controls);
+      verifyS3Exit(spans, httpEntry, withError);
+    }
+
+    function verifyHttpEntry(spans, apiPath, controls) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.p).to.not.exist,
+        span => expect(span.k).to.equal(constants.ENTRY),
+        span => expect(span.f.e).to.equal(String(controls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal('node.http.server'),
+        span => expect(span.data.http.url).to.equal(apiPath)
+      ]);
+    }
+
+    function verifyS3Exit(spans, parent, withError) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.n).to.equal('s3'),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.t).to.equal(parent.t),
+        span => expect(span.p).to.equal(parent.s),
+        span => expect(span.f.e).to.equal(String(appControls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.error).to.not.exist,
+        span => expect(span.ec).to.equal(withError ? 1 : 0),
+        span => expect(span.async).to.not.exist,
+        span => expect(span.data).to.exist,
+        span => expect(span.data.s3).to.be.an('object'),
+        span => expect(span.data.s3.op).to.exist
+      ]);
+    }
+  });
+
+  describe('tracing disabled', () => {
+    this.timeout(config.getTestTimeout() * 2);
+
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      tracingEnabled: false,
+      env: {
+        AWS_S3_BUCKET_NAME: bucketName
+      }
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
+
+    requestMethods.forEach(requestMethod => {
+      describe(`attempt to get result with: ${requestMethod}`, () => {
+        availableOperations.forEach(operation => {
+          it(`should not trace (${operation})`, async () => {
+            await appControls.sendRequest({
+              method: 'GET',
+              path: `/${operation}/${requestMethod}`
+            });
+            return retry(() => agentControls.getSpans(), retryTime)
+              .then(() => agentControls.getSpans())
```

Comment:
We need a delay here. Checking for the span immediately after sending the request does not prove we did not create any spans. The idea is:

* send the request
* wait for some time (say 500 ms) with `delay`
* then check for spans

If there are no spans after 500 ms, we can be reasonably sure that we did not create any spans.

Also, there are two agentControls.getSpans() calls here. The first one should be the delay instead.